action.go

```diff
@@ -295,21 +295,27 @@ func (c *cmdAction) ClusterCanRetry() bool {
 // set Nil to true. If not the return value will be unmarshaled into Rcv
 // normally.
 type MaybeNil struct {
-	Nil bool
-	Rcv interface{}
+	Nil   bool
+	Empty bool
+	Rcv   interface{}
 }
 
 // UnmarshalRESP implements the method for the resp.Unmarshaler interface.
 func (mn *MaybeNil) UnmarshalRESP(br *bufio.Reader) error {
 	var rm resp2.RawMessage
-	if err := rm.UnmarshalRESP(br); err != nil {
+	err := rm.UnmarshalRESP(br)
+	switch {
+	case err != nil:
 		return err
-	} else if rm.IsNil() {
+	case rm.IsNil():
 		mn.Nil = true
 		return nil
+	case rm.IsEmpty():
+		mn.Empty = true
+		return nil
+	default:
+		return rm.UnmarshalInto(resp2.Any{I: mn.Rcv})
```

Comment:
`UnmarshalInto` must be performed regardless of whether or not the response is an empty array, otherwise this breaks backwards compatibility.